compiler/rustc_middle/src/ty/print/mod.rs

```diff
@@ -128,16 +129,27 @@ pub trait Printer<'tcx>: Sized {
                 self.print_impl_path(def_id, substs, self_ty, impl_trait_ref)
             }
 
-            _ => {
+            def_path_data => {
                 let parent_def_id = DefId { index: key.parent.unwrap(), ..def_id };
 
+                // Inherent projections
+                if let DefPathData::TypeNs(_) = def_path_data
+                    && let DefKind::Impl { of_trait: false } = self.tcx().def_kind(parent_def_id)
+                    && let Some(self_ty) = substs.get(0).and_then(|subst| subst.as_type())
```

Comment:
In some cases the substs only contains consts or are empty which surprised me quite a bit.
This seems to happen when we print anon consts (which I'm not that familiar with).

E.g. for `<impl at dump_mir_1.rs:5:1: 5:31>::Mapping::{constant#0}`, the substs were empty or for
`Parent::<1>::Mapping::<{ 2 * 5 }>::{constant#0}` the substs were `[Const { ty: usize, kind: Value(Leaf(0x0000000000000001)) }, Const { ty: usize, kind: Value(Leaf(0x000000000000000a)) }]`.

I guess this is how anon consts are encoded. Their “encoding” shouldn't be able to clash with the
one of inherent projections though I would judge.

Reply:
Ah, ok. For the `[Const, Const]` case, the 1st const embodies the parent substs (here for `Parent`) and the 2nd one corresponds to the GAT substs. Makes sense